### PR TITLE
chore(ci): run tests in random order

### PR DESCRIPTION
**Pull request by justinchuby**
https://pypi.org/project/pytest-randomly/

Tests are still reproducible locally by supplying the random seed from CI:

image

> Randomness in testing can be quite powerful to discover hidden flaws in the tests themselves, as well as giving a little more coverage to your system.
>
> By randomly ordering the tests, the risk of surprising inter-test dependencies is reduced - a technique used in many places, for example Google’s C++ test runner googletest. Research suggests that “dependent tests do exist in practice” and a random order of test executions can effectively detect such dependencies [1]. Alternatively, a reverse order of test executions, as provided by pytest-reverse, may find less dependent tests but can achieve a better benefit/cost ratio.

### CODE REVIEW

**Comment by BowenBao** [comment]
Do our test cases have `det setUp` that fix the seed or no?

**Comment by justinchuby** [comment]
We do. the random seed for numpy and torch is still fixed as intended. Only the order of the test run is random

File: noxfile.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
     "pytest-cov",
     "pytest-subtests",
     "pytest-xdist",
+    "pytest-randomly",
```
Comment (jcwchen) [inline noxfile.py:25]: nit: make pytest-randomly in alphabetical order as below?

**Review by jcwchen — APPROVED** [review]

File: noxfile.py (hunk 1 above)
Comment (justinchuby) [reply noxfile.py]: Done!

**Pull request merged**